### PR TITLE
[10.0] stock_available_mrp : Improvements

### PR DESCRIPTION
**Pull request by rousseldenis**
@Cedric-Pigeon

### CODE REVIEW

File: stock_available_mrp/models/product_product.py [hunk 1]
```diff
@@ -16,40 +16,46 @@ class ProductProduct(models.Model):
...
-            product.bom_id = bom_by_product_id.get(
-                product.id,
-                bom_by_product_tmpl_id.get(product.product_tmpl_id.id)
+            product_boms = boms.filtered(
+                lambda b: b.product_id == product or
+                (not b.product_id and
+                 b.product_tmpl_id == product.product_tmpl_id)
             )
+            if product_boms:
+                product.bom_id = product_boms[0]
```
Comment (Cedric-Pigeon) [inline stock_available_mrp/models/product_product.py]: use `first()` from `from odoo.fields import first`

File: stock_available_mrp/models/product_product.py [hunk 2]
```diff
@@ -58,25 +64,18 @@ def _compute_available_quantities_dict(self):
...
         # explode all boms at once
         exploded_boms = product_with_bom._explode_boms()
 
         # extract the list of product used as bom component
-        product_components_ids = set()
+        component_products = self.env['product.product'].browse()
         for exploded_components in exploded_boms.values():
             for bom_component in exploded_components:
-                product_components_ids.add(bom_component[0].product_id.id)
+                component_products |= bom_component[0].product_id
```
Comment (Cedric-Pigeon) [inline stock_available_mrp/models/product_product.py]: Same here, better to use first()

**Review by Cedric-Pigeon — CHANGES_REQUESTED** [review]
Small remark otherwise sounds good to me.
Could you also check why travis is red ?

File: stock_available_mrp/models/product_product.py (hunk 1 above)
Comment (rousseldenis) [reply stock_available_mrp/models/product_product.py]: Ok, never saw this helper before... They could have put it in BaseModel...

**Comment by rousseldenis** [comment]
@Cedric-Pigeon I'll fix first().
Then, I'll fix all travis errors, but this seems to get linked to odoo changes in procurement behaviour I spoke to you yesterday. Will investigate

**Pull request merged**